contracts/nois-gateway/src/contract.rs

```diff
@@ -278,40 +424,76 @@ fn execute_set_config(
         Some(dr) => Some(deps.api.addr_validate(&dr)?),
         None => config.drand,
     };
-    let price = match price {
-        Some(pr) => pr,
-        None => config.price,
+    let price = price.unwrap_or(config.price);
+    let payment_code_id = match payment_code_id {
+        Some(new_code_id) => {
+            ensure_code_id_exists(deps.as_ref(), new_code_id)?;
+            new_code_id
+        }
+        None => config.payment_code_id,
     };
+    let payment_initial_funds = payment_initial_funds.unwrap_or(config.payment_initial_funds);
+
     let new_config = Config {
         manager,
         drand,
         price,
+        payment_code_id,
+        payment_initial_funds,
+        sink: config.sink, // Make updatable?
```

Comment:
Let's only make it updatable when we reach the stage where the manager is the  governance module.